activerecord/lib/active_record/fixtures.rb

```diff
@@ -520,53 +520,64 @@ def self.create_fixtures(fixtures_directory, fixture_set_names, class_names = {}
+          conn = klass ? klass.connection : connection
+          fixtures_map[fs_name] = new( # ActiveRecord::FixtureSet.new
               conn,
               fs_name,
               klass,
               ::File.join(fixtures_directory, fs_name))
-          end
+        end
 
-          update_all_loaded_fixtures fixtures_map
+        update_all_loaded_fixtures fixtures_map
 
-          connection.transaction(:requires_new => true) do
-            deleted_tables = Set.new
-            fixture_sets.each do |fs|
-              conn = fs.model_class.respond_to?(:connection) ? fs.model_class.connection : connection
-              table_rows = fs.table_rows
+        all_connections = ActiveRecord::Base.connection_handler.connection_pool_list.map(&:connection)
+        disable_referential_integrity_in_transaction(all_connections) do
+          deleted_tables = Set.new
+          fixture_sets.each do |fs|
+            conn = fs.model_class.respond_to?(:connection) ? fs.model_class.connection : connection
+            table_rows = fs.table_rows
 
-              table_rows.each_key do |table|
-                unless deleted_tables.include? table
-                  conn.delete "DELETE FROM #{conn.quote_table_name(table)}", 'Fixture Delete'
-                end
+            table_rows.each_key do |table|
+              unless deleted_tables.include? table
+                conn.delete "DELETE FROM #{conn.quote_table_name(table)}", 'Fixture Delete'
                 deleted_tables << table
               end
-
-              table_rows.each do |fixture_set_name, rows|
-                rows.each do |row|
-                  conn.insert_fixture(row, fixture_set_name)
-                end
+            end
+            table_rows.each do |fixture_set_name, rows|
+              rows.each do |row|
+                conn.insert_fixture(row, fixture_set_name)
               end
+            end
 
-              # Cap primary key sequences to max(pk).
-              if conn.respond_to?(:reset_pk_sequence!)
-                conn.reset_pk_sequence!(fs.table_name)
-              end
+            # Cap primary key sequences to max(pk).
+            if conn.respond_to?(:reset_pk_sequence!)
+              conn.reset_pk_sequence!(fs.table_name)
             end
           end
-
-          cache_fixtures(connection, fixtures_map)
         end
+
+        cache_fixtures(connection, fixtures_map)
       end
       cached_fixtures(connection, fixture_set_names)
     end
 
+    # disables referential integrity on all connections and
+    # starts a transaction on all connection. Finally it executes the
+    # given block.
+    def self.disable_referential_integrity_in_transaction connections, &block
+      return block.call if connections.empty?
+      connection = connections.shift
+      connection.disable_referential_integrity do
+        connection.transaction(:requires_new => true) do
+          disable_referential_integrity_in_transaction connections, &block
```

Comment:
This is the only way (I could come up with) do `disable_referential_integrity` AND `transaction` using the block syntax. 
Iterative approach using `each` wouldn't work here.

I think it's easier to read/understand than that pseudo code:
```
connections.each do |connection|
begin
  connection.disable_referential_integrity
  connection.start_transaction
  yield
  connection.check_if_rollback ? connection.rollback : connection.commit
  connection.enable_referential_integrity
rescue SomeExceptions
ensure
  other stuff
end
```